configman/config_manager.py

```diff
@@ -639,6 +639,7 @@ def _overlay_expand(self):
                         # the value source.  This assignment may come
                         # via acquisition, so the key given may not have
                         # been an exact match for what was returned.
+                        opt.changed_flag = opt.default != val_src_dict[key]
```

Comment:
good catch.  obviously the use of this flag is not covered by a test in the ConfigurationManager.